src/store/actions/UserProfile.ts

```diff
@@ -0,0 +1,20 @@
+import { ReduxAction } from "types/redux";
+
+import { User } from "types/User";
+import { WithId } from "utils/id";
+
+export enum UserProfileActionTypes {
+  SET_USER_PROFILE_DATA = "SET_USER_PROFILE_DATA",
+}
+
+export type SetUserProfileDataAction = ReduxAction<
+  UserProfileActionTypes.SET_USER_PROFILE_DATA,
+  { userProfile?: WithId<User> }
+>;
+
+export const setUserProfileData = (userProfile?: WithId<User>) => ({
+  type: UserProfileActionTypes.SET_USER_PROFILE_DATA,
+  payload: { userProfile },
+});
```

Comment:
My previous review comments in https://github.com/sparkletown/sparkle/pull/1218#discussion_r585248664 were marked as resolved, yet seemingly nothing ever actually done about them? Why were they marked as resolved?